test/v2/unit/coins/algo.ts

```diff
@@ -0,0 +1,118 @@
+import { strict as assert } from 'assert';
+import * as Promise from 'bluebird';
+const co = Promise.coroutine;
+const Wallet = require('../../../../src/v2/wallet');
+
+import 'should';
+
+const TestV2BitGo = require('../../../lib/test_bitgo');
+
+const nock = require('nock');
+nock.enableNetConnect();
+
+describe('ALGO:', function() {
+  let bitgo;
+  let basecoin;
+
+  before(function() {
+    bitgo = new TestV2BitGo({ env: 'test' });
+    bitgo.initializeTestVars();
+    basecoin = bitgo.coin('talgo');
+  });
+
+  after(function() {
+    nock.cleanAll();
+  });
+
+  describe('Transaction Verification', function() {
+    let basecoin;
+    let wallet;
+    let halfSignedTransaction;
+    let prebuild;
+
+    const userKeychain = {
+      pub: 'UMYEHZ2NNBYX43CU37LMINSHR362FT4GFVWL6V5IHPRCJVPZ46H6CBYLYE',
+      prv: 'HEHPVOKEINTQMW4K536BZWUZWQNKBXKIAUSEWCVX3LCXDHI2LVPMV6T5L4'
+    };
+
+    const backupKeychain = {
+      pub: 'LT7DY6JX46K2MUCLJJHEUYWKVC63FXEKCOKUSPGVTDGH3DBBK2V4SEDZWY',
+      prv: 'R4MFSNIAR4PQQGGYA6LK374X6MJEATRVNLREZ3GLAP2VYWLXDN2R4JLI4'
+    };
+
+    const bitgoKeychain = {
+      pub: '3GFMZSPBX3XQL22IBAUF2UQFAZCOF5QFFVRRLEHFPHFLN5NY7WQJSSZRGQ',
+      prv: 'DLHE27PAI335VEQUNX7R2442SSJZUYISBL7OOR3V34AYNTH7YWI6XMGVVE'
+    }
+
+    const signedTxBase64 = 'AAAAAGRnXg19FteG/7zPd+jDC7LDvRlzgfFC+JrPhRep0kYiAAAAZAB/4cUAAAACAAAAAAAAAAAAAAABAAAAAQAAAABkZ14NfRbXhv+8z3fowwuyw70Zc4HxQviaz4UXqdJGIgAAAAEAAAAAmljT/+FedddnAHwo95dOC4RNy6eVLSehaJY34b9GxuYAAAAAAAAAAAehIAAAAAAAAAAAAUrgwAkAAABAOExcvVJIUJv9HuVfbV0y7lRPRARv4wDtcdhHG7QN40h5wQ2uwPF52OGQ8KY+66a1A/8lNKB75sgj2xj44s8lDQ==';
+
+    before(co(function *() {
+      basecoin = bitgo.coin('talgo');
+      const walletData = {
+        id: '5a78dd561c6258a907f1eeaee132f796',
+        users: [
+          {
+            user: '543c11ed356d00cb7600000b98794503',
+            permissions: [
+              'admin',
+              'view',
+              'spend'
+            ]
+          }
+        ],
+        coin: 'talgo',
+        label: 'Test Validation Wallet',
+        m: 2,
+        n: 3,
+        keys: [
+          '5a78dd56bfe424aa07aa068651b194fd',
+          '5a78dd5674a70eb4079f58797dfe2f5e',
+          '5a78dd561c6258a907f1eea9f1d079e2'
+        ],
+        tags: [
+          '5a78dd561c6258a907f1eeaee132f796'
+        ],
+        disableTransactionNotifications: false,
+        freeze: {},
+        deleted: false,
+        approvalsRequired: 1,
+        isCold: true,
+        coinSpecific: {},
+        clientFlags: [],
+        balance: 650000000,
+        confirmedBalance: 650000000,
+        spendableBalance: 650000000,
+        balanceString: '650000000',
+        confirmedBalanceString: '650000000',
+        spendableBalanceString: '650000000',
+        receiveAddress: {
+          id: '5a78de2bbfe424aa07aa131ec03c8dc1',
+          address: 'GBSGOXQNPULNPBX7XTHXP2GDBOZMHPIZOOA7CQXYTLHYKF5J2JDCF7LT',
+          chain: 0,
+          index: 0,
+          coin: 'txlm',
```

Comment:
talgo

Reply:
k. I changed all this. this mock wallet was copied from something else